pyquil/api/_qam.py

```diff
@@ -60,18 +61,36 @@ def load(self, executable):
         return self
 
     @_record_call
-    def write_memory(self, *, region_name: str, offset: int = 0, value=None):
+    def write_memory(self, *, region_name: str, offset: Optional[int] = None,
+                     value: Optional[Union[int, float, Sequence[int], Sequence[float]]] = None):
         """
-        Writes a value into a memory region on the QAM at a specified offset.
+        Writes a value or chronologically unwraps a list of values into a memory region on
+        the QAM at a specified offset.
 
         :param region_name: Name of the declared memory region on the QAM.
         :param offset: Integer offset into the memory region to write to.
-        :param value: Value to store at the indicated location.
+        :param value: Value(s) to store at the indicated location.
         """
         assert self.status in ['loaded', 'done']
 
-        aref = ParameterAref(name=region_name, index=offset)
-        self._variables_shim[aref] = value
+        if offset is None:
+            offset = 0
```

Comment:
there was a question on the previous PR about why this should be `0` -- technically we could insert a values list at some offset, but that's not really the intended reason for this change. this `write_memory` function is really only used from the top-level `QuantumComputer` object and when a list of memory register values is passed to the QC's methods, it is assumed that the list starts at offset 0. thus, that is why `offset` is set to 0 when we're passing a list